### PR TITLE
feat: add hash property

### PR DESCRIPTION
**Pull request by Aman-zishan**
- changes made to importer and exporter contracts

### Pull-Request Checklist

- [x] Code is up-to-date with the `main` branch
- [ ] `yarn lint` passes with this change
- [ ] `yarn clarinet:check` passes with this change
- [ ] `yarn clarinet:tests` passes with this 
- [ ] `yarn unit-tests` passes with this
- [ ] Run `yarn eol`
- [ ] Run `yarn format`
- [ ] Run `yarn cleanup`
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [ ] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: packages/clarity/contracts/taral/exporter-storage.clar [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
-(define-map exporter-profile {exporter-id: uint} 
-  {
-    name: (string-utf8 100), 
-    category: (string-utf8 100), ;; Merchant /Manufacturer /Service /Project /Deemed exporter
-    orders-next-avail-id: uint
-  }  
+(define-map exporter-profile 
+    {
+        exporter-id: uint,
+        hash: (buff 256)
```
Comment (dorucioclea) [inline packages/clarity/contracts/taral/exporter-storage.clar:10]: Nah, the hash needs to be added in the value side of the map. Two different hashes of the same exporter just mean that the exporter  has changed, but it is still the same exporter. In your case, the `exporter-profile` map could consider the same exporter that has some fields changed as a new exporter, hence, the hash property does not need to be placed in the `key` part of the exporter profile map

**Review by dorucioclea — CHANGES_REQUESTED** [review]
As described, the hash should not be part of the key, this triggers other changes, probably where the hash is set/get in the set/get functions

File: packages/clarity/contracts/taral/exporter-storage.clar [hunk 2]
```diff
@@ -62,32 +77,42 @@
     (ok (map-set exporter-by-principal exporter exporter-id))
 )
 
-(define-public (add-order (id uint) (exporter-id uint) (order-id uint)) 
+(define-public (add-order (id uint) (exporter-id uint) (hash (buff 256)) (order-id uint)) 
     (ok 
         (map-insert orders {id: id,exporter-id: exporter-id}   
-            {order-id: order-id }     
+            {order-id: order-id, hash: hash }
```
Comment (dorucioclea) [inline packages/clarity/contracts/taral/exporter-storage.clar:83]: O think particularly for orders, we do not need to keep the hash property here as this is just a map that links exporters to their orders. Apologies for any confusion 😁

File: packages/clarity/contracts/taral/exporter-storage.clar [hunk 3]
```diff
@@ -62,32 +77,42 @@
...
+                created: block-height
+            }
         )
     )
 )
 
 ;; @Desc function to to filter out none from list [none none (some XXX) none]
 ;; @Param value : tuple containing the order ID
-(define-private (is-valid-value (value (optional {order-id: uint})))
+(define-private (is-valid-value (value (optional {order-id: uint, hash: (buff 256)})))
```
Comment (dorucioclea) [inline packages/clarity/contracts/taral/exporter-storage.clar:116]: As a consequence of my prev comment, hash can be dropped here

File: packages/clarity/contracts/taral/importer-storage.clar [hunk 4]
```diff
@@ -1,18 +1,32 @@
...
 
-(define-map orders {id: uint,importer-id: uint} {order-id: uint})
+(define-map orders 
+    {
+        id: uint,
+        importer-id: uint,
+    } 
+    {
+        order-id: uint, 
+        hash: (buff 256)
```
Comment (dorucioclea) [inline packages/clarity/contracts/taral/importer-storage.clar:27]: Again, we don't need to keep the hash property here

**Review by dorucioclea — CHANGES_REQUESTED** [review]
Nearly there ✌️

File: packages/clarity/contracts/taral/exporter-storage.clar (hunk 2 above)
Comment (Aman-zishan) [reply packages/clarity/contracts/taral/exporter-storage.clar:83]: Makes sense, i'll remove the hash property from here.

File: packages/clarity/contracts/taral/taral-exporter-v1.clar [hunk 5]
```diff
@@ -38,35 +54,48 @@
...
-                (new-id (contract-call? .exporter-storage get-orders-next-avail-id current-exporter))
-                )
-            (unwrap! (contract-call? .exporter-storage update-exporter-profile {exporter-id: exporter-id} (merge current-exporter { orders-next-avail-id: (+ u1 new-id)})) exporter-storage-error)
-            (unwrap! (contract-call? .exporter-storage add-order new-id exporter-id new-order-id) exporter-storage-error)
-            (print {action: "append-order", exporter: exporter, new-order-id: new-order-id  })
-            (ok true)
-            )
+        ;; check that the hash is not empty
+        (asserts! (> (len hash) u0) ERR_EMPTY_HASH)
+        (unwrap! (contract-call? .exporter-storage update-exporter-profile {exporter-id: exporter-id} (merge current-exporter { orders-next-avail-id: (+ u1 new-id)})) exporter-storage-error)
```
Comment (Aman-zishan) [inline packages/clarity/contracts/taral/taral-exporter-v1.clar:94]: @dorucioclea do u think we should merge the hash property here along with incrementing `orders-next-avail-id` cause we are updating the exporter profile and that means change in data right. 

```suggestion
        (unwrap! (contract-call? .exporter-storage update-exporter-profile {exporter-id: exporter-id} (merge current-exporter { orders-next-avail-id: (+ u1 new-id), hash: hash })) exporter-storage-error)
```

**Comment by Aman-zishan** [comment]
### CHANGELOG: `taral-purchase-order.clar` -> `taral-purchase-order-v1.clar`

- Variables removed from `order` onchain data
    - delivery country
    - dispatch method
    - shiptment type

- Variables removed from `order-details` onchain data
    - shippping Route list
    - item list

- Removed nested `let` statements and refactored `initialize` function to handle all operations under one `let` statement

### purchase-order-v1 data mappings before and after

Screenshot 2023-07-10 at 7 30 20 PM
Screenshot 2023-07-10 at 7 24 49 PM

**Comment by Aman-zishan** [comment]
Closing this as commits replicated in #1301

**Pull request closed without merging**